src/app/revisors/permissions.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { TokenService } from '../services/token.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PermissionsService implements CanActivate{
+
+  rol: string;
+
+  constructor(
+    private tokenService: TokenService,
+    private router: Router
+  ) { }
+
+  
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean{
+    const expectedRol = route.data.expectedRol;
+    const roles = this.tokenService.getAuthorities();
+    this.rol = 'user';
+
+    roles.forEach( rol => {
+      if(rol == 'ROLE_ADMIN'){
+        this.rol = 'admin'
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ispp-yourney/angular-app/pullRequest?prid=7122565)
```suggestion
        this.rol = "admin"
```